dev/system-test/firestore.ts

```diff
@@ -1647,12 +1683,61 @@ describe('Query class', () => {
       });
   });
 
-  it('has limit() method', async () => {
+  it('can run get() on empty collection', async () => {
+    return randomCol.get().then(res => {
+      return expect(res.empty);
+    });
+  });
+
+  it('can run stream() on empty collection', async () => {
+    let received = 0;
+    const stream = randomCol.stream();
+
+    for await (const doc of stream) {
+      expect(doc).to.be.an.instanceOf(QueryDocumentSnapshot);
+      ++received;
+    }
+
+    expect(received).to.equal(0);
+  });
+
+  it('has limit() method on get()', async () => {
     await addDocs({foo: 'a'}, {foo: 'b'});
     const res = await randomCol.orderBy('foo').limit(1).get();
     expectDocs(res, {foo: 'a'});
   });
 
+  it('has limit() method on stream()', async () => {
+    let received = 0;
+    await addDocs({foo: 'a'}, {foo: 'b'}).then(async () => {
+      const stream = randomCol.orderBy('foo').limit(1).stream();
+      for await (const doc of stream) {
+        expect(doc).to.be.an.instanceOf(QueryDocumentSnapshot);
+        ++received;
+      }
+
+      expect(received).to.equal(1);
+    });
```

Comment:
Nit: since you have an async test, you can drop the use of `promise.then(...)` and make the code even more readable.

```
  await addDocs({foo: 'a'}, {foo: 'b'});

  const stream = randomCol.orderBy('foo').limit(1).stream();
  for await (const doc of stream) {
    expect(doc).to.be.an.instanceOf(QueryDocumentSnapshot);
    ++received;
  }

  expect(received).to.equal(1);
```